### PR TITLE
fix: Ptt Multi Forward(8.9.88-4852)

### PR DESCRIPTION
**Pull request by HdShare**
# fix: Ptt Multi Forward(8.9.88-4852)

## Description
适配 8.9.88 语音多选转发

## Issues Fixed or Closed by This PR
#106 

## Check List
- [x] I have tested the changes and verified that they work and don't break anything(as well as I can manage) or drop the support for previous versions.
- [x] I have merged commits that are meaningless for follow-up work and confirmed that they will not cause damage to follow-up maintenance

### CODE REVIEW

File: app/src/main/java/cc/ioctl/hook/msg/PttForwardHook.java [hunk 1]
```diff
@@ -276,12 +276,22 @@ protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
                     if (targets.size() > 1) {
                         multi = true;
                         for (Object rr : targets) {
-                            ContactDescriptor c = ContactDescriptor.parseResultRec(rr);
+                            ContactDescriptor c;
```
Comment (cinit) [inline app/src/main/java/cc/ioctl/hook/msg/PttForwardHook.java:279]: 建议不要在这里加 if, 每次用都要判断很麻烦，建议直接修改 `ContactDescriptor.parseResultRec`

File: app/src/main/java/io/github/qauxv/util/data/ContactDescriptor.java [hunk 2]
```diff
@@ -28,7 +28,7 @@
 public class ContactDescriptor {
 
     public String uin;
-    public int uinType;
+    public Integer uinType;
```
Comment (cinit) [inline app/src/main/java/io/github/qauxv/util/data/ContactDescriptor.java:31]: 请说明一下为什么需要将 int 改为 Integer

File: app/src/main/java/io/github/qauxv/util/data/ContactDescriptor.java [hunk 3]
```diff
@@ -49,4 +49,12 @@ public static ContactDescriptor parseResultRec(Object a) {
         cd.uinType = Reflex.getInstanceObjectOrNull(a, "b", int.class);
         return cd;
     }
+
+    public static ContactDescriptor parseResultRec_QQ_8_9_88(Object a) {
```
Comment (cinit) [inline app/src/main/java/io/github/qauxv/util/data/ContactDescriptor.java:53]: 建议直接改 parseResultRec 没有必要单独开一个方法

File: app/src/main/java/io/github/qauxv/util/data/ContactDescriptor.java (hunk 2 above)
Comment (HdShare) [reply app/src/main/java/io/github/qauxv/util/data/ContactDescriptor.java:31]:
```java
val uinType = Reflex.getInstanceObjectOrNull(obj, "uinType", int.class);
```
因为 uinType 可能会出现null的情况

使用int uinType=null
会抛出空指针异常

使用Integer uinType=null
使用接收就不会抛出异常

File: app/src/main/java/io/github/qauxv/util/data/ContactDescriptor.java (hunk 3 above)
Comment (HdShare) [reply app/src/main/java/io/github/qauxv/util/data/ContactDescriptor.java:53]: > 建议直接改 parseResultRec 没有必要单独开一个方法

在parseResultRec里面进行判断版本?ok，我改改

**Pull request closed without merging**